src/common/common.cpp

```diff
@@ -774,28 +774,30 @@ bool find_app_name_in_path(const std::wstring& where, const std::vector<std::wst
     return false;
 }
 
-std::optional<std::string> exec_and_read_output(const std::wstring_view command, const DWORD timeout)
+std::optional<std::string> exec_and_read_output(const std::wstring_view command, DWORD timeout_ms)
 {
     SECURITY_ATTRIBUTES saAttr{ sizeof(saAttr) };
-    saAttr.bInheritHandle = true;
+    saAttr.bInheritHandle = false;
 
-    wil::unique_handle childStdoutRead;
-    wil::unique_handle childStdoutWrite;
-    if (!CreatePipe(&childStdoutRead, &childStdoutWrite, &saAttr, 0))
-    {
-        return std::nullopt;
-    }
+    constexpr size_t bufferSize = 4096;
+    // We must use a named pipe for async I/O
+    char pipename[MAX_PATH + 1];
+    GetTempFileNameA(R"(\\.\pipe\)", "tmp", 1, pipename);
+    wil::unique_handle readPipe{ CreateNamedPipeA(pipename, PIPE_ACCESS_INBOUND | FILE_FLAG_OVERLAPPED, PIPE_TYPE_BYTE | PIPE_READMODE_BYTE, PIPE_UNLIMITED_INSTANCES, bufferSize, bufferSize, 0, &saAttr) };
```

Comment:
Same for CreateNamedPipeA, check the return value.

Reply:
that's done a bit below - `    if (!readPipe || !writePipe)`